### ISSUE TITLE
[BUG] Running test in a docker container in circleCI is terminating before report generation

### ISSUE THREAD

**Issue by amrsa1** [microsoft/playwright#22189]
### System info
- Playwright Version:1.32.2
- Operating System: linux
- Browser: [All, Chromium, Firefox, WebKit]
- Other info:

### Source code

            docker-compose up --exit-code-from e2e--build e2e

**Steps**
- Run a test in a container in circleCI

**Expected**

Delay exit code after the report generation might solve the issue

**Actual**

<img width="731" alt="Screenshot 2023-04-04 at 09 45 34" src="https://user-images.githubusercontent.com/32042806/229803221-5993f7d9-41cf-45c3-8336-514516b27641.png">

In order to show step failure we have to pass --exit-code-from  flag, but this flag will case to terminate the container right away once test is done and playwright return exit code 1, this will case container to terminate before report generating

**Comment by pavelfeldman** [comment]
Test runner exits after the report has been generated. If you need to wait for something after Playwright is done, you need to do that yourself since we have no way of knowing what we are waiting for.

**Comment by amrsa1** [comment]
But in the log it shows that the reported wasnt generated, no ?

**Comment by pavelfeldman** [comment]
Which log?

**Comment by amrsa1** [comment]
I meant the screenshot in description, i dont see the part that report has been generated.

**Comment by pavelfeldman** [comment]
Playwright does not print any messages about reports being generated. Playwright is just an application. Are you printing some message that is not there? I'm seeing a failed test run on the screenshot and it looks like Playwright did everything as documented: issued report callbacks (hence the reporter epilogue on the screenshot) and exited with status code 1 as seen on the screenshot.

Since the thread is getting long and we don't seem to be getting closer to the culprit, it'll be hard for others to follow. Could you please file a new issue where you follow the issue template and provide exact steps to be performed along with your expectations.
